include/deal.II/base/mpi.h

```diff
@@ -952,6 +967,28 @@ namespace Utilities
 #  endif
     }
 
+
+#  ifdef DEAL_II_WITH_MPI
+    template <class Iterator, typename Number>
+    std::pair<Number, typename numbers::NumberTraits<Number>::real_type>
```

Comment:
Couldn't you figure out `Number` as `decltype(*declval(Iterator))`?

Reply:
> Couldn't you figure out Number as decltype(*declval(Iterator))?

nope, that's not good for integers.

Reply:
Oh, you're saying that you want to put integers in and get doubles out? I see. Could you document this use case with a `@tparam`?

Reply:
> Oh, you're saying that you want to put integers in and get doubles out?

yeap, see the tests.

> Could you document this use case with a @tparam?

will do.